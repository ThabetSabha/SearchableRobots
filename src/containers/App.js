import React, { Component } from 'react';
import CardList from '../components/CardList';
import Searchbox from '../components/Searchbox';
import './App.css';
import { connect } from 'react-redux';
import { requestRobotsAction, setSearchField } from '../actions';


const mapStateToProps = (state) => {
    return {
        searchField: state.searchRobots.searchField,
        robots: state.requestRobotsReducer.robots,
        isPending: state.requestRobotsReducer.isPending,
        error: state.requestRobotsReducer.error
    };

}


const mapDispatchToProps = (dispatch) => {
    return {
        onSearchChange: (event) => dispatch(setSearchField(event.target.value)),
        onRequestRobots: () => requestRobotsAction(dispatch)
    }
}

class App extends Component {
   
    componentDidMount() {
            this.props.onRequestRobots();
    }


    render() {
        //Appearntly we can't have const or define variables that aren't functions 
        //unless it's under render.
        
        const { onSearchChange, searchField,robots,isPending } = this.props;
        const filtered = robots.filter(robot => {
            return robot.name.toLowerCase().includes(searchField.toLowerCase());
        })

        if (isPending) {
            return (
                <div className="tc">
                    <h1 className="ma2 f1 ">RoboFriends</h1>
                    <Searchbox SearchChange={onSearchChange} />
                    <div>
                        <h1 className="ma2 f1 ">Loading..</h1>
                    </div>
                </div>
            )
        } else {

            return (
                <div className="tc main">
                    <h1 className="ma2 f1 ">RoboFriends</h1>
                    <Searchbox SearchChange={onSearchChange} />
                    <div style={{ overflowY: 'scroll', height: '500px', border: '2px solid grey' }}>
                        <CardList robots={filtered} />
                    </div>
                </div>
            )
        }
    }
}


export default connect(mapStateToProps, mapDispatchToProps)(App);