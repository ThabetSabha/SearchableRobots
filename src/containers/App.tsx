import React, { useEffect } from "react";
import CardList from "../components/CardList/CardList";
import Searchbox from "../components/SearchBox/Searchbox";
import "./App.css";
import { RootStateOrAny, useDispatch, useSelector } from "react-redux";
import { requestRobotsAction, setSearchField } from "../redux/actions";

const App = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(requestRobotsAction());
  }, [dispatch]);

  const onSearchChange = (event: React.SyntheticEvent): void => {
    const element = event.target as HTMLInputElement;
    dispatch(setSearchField(element.value));
  };

  const searchField = useSelector(
    (state: RootStateOrAny) => state.searchRobots.searchField
  );
  const robots = useSelector(
    (state: RootStateOrAny) => state.requestRobotsReducer.robots
  );
  const isPending = useSelector(
    (state: RootStateOrAny) => state.requestRobotsReducer.isPending
  );

  const filtered = robots.filter((robot: IRobot) => {
    return robot.name.toLowerCase().includes(searchField.toLowerCase());
  });

  return (
    <div className="tc main">
      <h1 className="ma2 f1 ">RoboFriends</h1>
      <Searchbox searchChange={onSearchChange} />
      <div
        style={{
          overflowY: "scroll",
          height: "500px",
          border: "2px solid grey",
        }}
      >
        {isPending ? (
          <h1 className="ma2 f1 ">Loading..</h1>
        ) : (
          <CardList robots={filtered} />
        )}
      </div>
    </div>
  );
};

export default App;
