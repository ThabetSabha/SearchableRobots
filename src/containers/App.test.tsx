import React from "react";
import App from "./App";
import { Provider } from "react-redux";
import { mount } from "enzyme";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";

const mockStore = configureMockStore([thunk]);
describe("Testing App component", () => {
    test("That the App Component Renders", () => {
        const store = mockStore({
          searchRobots: {
            searchField: "",
          },
          requestRobotsReducer: {
              isPending: false,
              robots: [],
              error: "",
          },
        });
      
        const wrapper = mount(
          <Provider store={store}>
            <App />
          </Provider>
        );
        expect(wrapper).toMatchSnapshot();
      });
    
})
