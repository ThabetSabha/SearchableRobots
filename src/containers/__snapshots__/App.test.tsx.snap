// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App component That the App Component Renders 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <App>
    <div
      className="tc main"
    >
      <h1
        className="ma2 f1 "
      >
        RoboFriends
      </h1>
      <Searchbox
        searchChange={[Function]}
      >
        <input
          aria-label="search Robots"
          className="pa3 ma4 ba b--green bg-lightest-blue robots-search-input"
          onChange={[Function]}
          placeholder="Search Robots"
          type="search"
        />
      </Searchbox>
      <div
        style={
          Object {
            "border": "2px solid grey",
            "height": "500px",
            "overflowY": "scroll",
          }
        }
      >
        <CardList
          robots={Array []}
        />
      </div>
    </div>
  </App>
</Provider>
`;
